elisp/ghc-check.el

```diff
@@ -65,7 +65,7 @@ nil            does not display errors/warnings.
 (defun ghc-check-syntax ()
   (interactive)
   (ghc-with-process (ghc-check-send)
-		    'ghc-check-callback
+            `(lambda (status) (interactive) (ghc-check-callback ,(file-name-directory (buffer-file-name)) status))
```

Comment:
I don't think that `(interactive)` is neede here since that function can't be called using M-x anyways since it doesn't have a name ;)